spec/features/user_edits_book_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+feature 'User edits an existing book' do
+  scenario 'updates title and author field successfully' do
+    book = create(:book)
+    visit root_path
+
+    edit_book(new_title: 'The Golden Compass', new_author: 'Philip Pullman')
+
+    expect(page.current_path).to eq books_path
+    expect(page).to have_content('The Golden Compass')
+    expect(page).to have_content('Philip Pullman')
+  end
```

Comment:
You may also want to add a scenario that tests when an update fails. For example, when a user passes an empty title `''`